src/sentry/snuba/discover.py

```diff
@@ -0,0 +1,230 @@
+from __future__ import absolute_import
+
+import six
+
+from copy import deepcopy
+from sentry.api.event_search import get_filter, resolve_field_list
+from sentry.utils.snuba import (
+    Dataset,
+    DISCOVER_COLUMN_MAP,
+    QUOTED_LITERAL_RE,
+    get_function_index,
+    raw_query,
+)
+
+ALLOWED_INTERNALS = set(["project_id", "event_id"])
+
+
+def resolve_column(col):
+    """
+    Resolve a public schema name to the discover dataset.
+    unknown columns are converted into tags expressions.
+    """
+    if not col:
+        return ""
+    if col.startswith("tags[") or col in ALLOWED_INTERNALS or QUOTED_LITERAL_RE.match(col):
+        return col
+    return DISCOVER_COLUMN_MAP.get(col, u"tags[{}]".format(col))
+
+
+def resolve_condition(cond):
```

Comment:
This doesn't seem to work generically for the CNF expressions of conditions as presented on snuba's README.

For example

```
        assert discover.resolve_condition(
            [[["issue.id", "=", "1"], ["x", "=", "2"]], [["y", "=", "3"], ["z", "=", "4"]]]
        ) == [[["group_id", "=", "1"], ["x", "=", "2"]], [["y", "=", "3"], ["z", "=", "4"]]]
```

Reply:
Nevermind. `discover.resolve_condition` actually resolves `OR` level conditions.